test/anyone-can-renew-test.js

```diff
@@ -0,0 +1,331 @@
+
+    // Have to add this last because wallet fund clears existing input data
+    const witness = new Stack();
+    witness.pushData(script.encode());
+    mtx.inputs[0].witness.fromStack(witness);
+
+    assert.throws(
+      () => mtx.check(),
+      {message: 'EVAL_FALSE'}
+    );
+  });
+
+  it('should fail to spend without correct action type: TRANSFER', async () => {
+    const mtx = new MTX();
+    mtx.addCoin(coin);
+
+    mtx.addOutput(new Output({
+      value: coin.value,
+      address: coin.address
+    }));
+    mtx.output(0).covenant.type = rules.types.TRANSFER;
+    mtx.output(0).covenant.pushHash(nameHash);
+    mtx.output(0).covenant.pushU32(heightBeforeOpen + 1);
+    mtx.output(0).covenant.pushU8(0);
+    mtx.output(0).covenant.push(Buffer.alloc(20));
+
+    await alice.fund(mtx, {coins: [coin]});
+    await alice.finalize(mtx, {coins: [coin]});
+    await alice.sign(mtx);
+
+    // Have to add this last because wallet fund clears existing input data
+    const witness = new Stack();
+    witness.pushData(script.encode());
+    mtx.inputs[0].witness.fromStack(witness);
+
+    assert.throws(
+      () => mtx.check(),
+      {message: 'EVAL_FALSE'}
+    );
+  });
+
+  it('should spend with correct action type: RENEW - Alice', async () => {
+    const mtx = new MTX();
+    mtx.addCoin(coin);
+
+    mtx.addOutput(new Output({
+      value: coin.value,
+      address: coin.address
+    }));
+    mtx.output(0).covenant.type = rules.types.RENEW;
+    mtx.output(0).covenant.pushHash(nameHash);
+    mtx.output(0).covenant.pushU32(heightBeforeOpen + 1);
+    mtx.output(0).covenant.pushHash(node.chain.tip.hash);
+
+    await alice.fund(mtx, {coins: [coin]});
+    await alice.finalize(mtx, {coins: [coin]});
+    await alice.sign(mtx);
+
+    // Have to add this last because wallet fund clears existing input data
+    const witness = new Stack();
+    witness.pushData(script.encode());
+    mtx.inputs[0].witness.fromStack(witness);
+
+    mtx.check();
+  });
+
+  it('should spend with correct action type: RENEW - Bob', async () => {
+    const mtx = new MTX();
+    mtx.addCoin(coin);
+
+    mtx.addOutput(new Output({
+      value: coin.value,
+      address: coin.address
+    }));
+    mtx.output(0).covenant.type = rules.types.RENEW;
+    mtx.output(0).covenant.pushHash(nameHash);
+    mtx.output(0).covenant.pushU32(heightBeforeOpen + 1);
+    mtx.output(0).covenant.pushHash(node.chain.tip.hash);
+
+    await bob.fund(mtx, {coins: [coin]});
+    await bob.finalize(mtx, {coins: [coin]});
+    await bob.sign(mtx);
+
+    // Have to add this last because wallet fund clears existing input data
+    const witness = new Stack();
+    witness.pushData(script.encode());
+    mtx.inputs[0].witness.fromStack(witness);
+
+    mtx.check();
+
+    // Bob (aka "anyone") will broadcast
+    const heightBeforeRenewal = node.chain.height;
+    await node.sendTX(mtx.toTX());
+    await mineBlocks(1);
+
+    const ns = await node.getNameStatus(nameHash);
```

Comment:
Maybe test that the name resource i.e TXT record is preserved here.

Reply:
👍